src/components/workspace-utils.js

```diff
@@ -56,6 +58,37 @@ export const WorkspaceSelector = ({ workspaces, value, onChange, ...props }) =>
   })
 }
 
+export const SnapshotInfo = ({ snapshotId, snapshotName }) => {
+  const [snapshotInfo, setSelectedSnapshotInfo] = useState({})
+  const signal = Utils.useCancellation()
+
+  Utils.useOnMount(() => {
+    const loadSnapshotInfo = async () => {
+      const snapshotInfo = await Ajax(signal).DataRepo.snapshots().snapshot(snapshotId).details()
+      setSelectedSnapshotInfo(snapshotInfo)
+    }
+
+    loadSnapshotInfo()
+  })
+
+  const { name, description, createdDate } = snapshotInfo
+
+  return _.isEqual(snapshotInfo, {}) ? spinnerOverlay : div({
+    style: { ...Style.elements.card.container }
```

Comment:
nit: The spread is unnecessary here, we could just reference the existing object directly.